File: 3499/solution.java
class Solution {
    public int maxActiveSectionsAfterTrade(String s) {
        int ones = 0;
        int prev = Integer.MIN_VALUE;
        int curr = 0;
        int res = 0;
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='0')
                curr++;
            else{
                if(curr>0){
                    res = Math.max(res, curr+prev);
                    prev = curr;
                    curr = 0;
                }
                ones++;
            }
        }
        if(curr>0)
            res = Math.max(res, curr+prev);
        res+=ones;        
        return res;
    }
}

File: 3499/solution.py
class Solution:
    def maxActiveSectionsAfterTrade(self, s: str) -> int:
        ones = 0
        curr = 0
        prev = float('-inf')
        res = 0
        for i in s:
            if i=='0':
                curr+=1
            else:
                if curr:
                    res = max(res, curr+prev)
                    prev = curr 
                    curr = 0
                ones+=1    
        if curr:
            res = max(res, curr+prev)
        
        res+=ones
        return res
